src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioDotNetObjectWrapper.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Reflection;
+using Microsoft.PowerFx;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Public.Values;
+using Xunit;
+
+namespace Microsoft.PowerFx.Interpreter.Tests
+{
+    // Demonstrate lazy dynamic marshalling 
+    public class ScenarioDotNetObjectWrapper
+    {
+        private class TestObj
+        {
+            public int Value { get; set; }
+
+            // Recursion, but handled since we're dynamically marshalling. 
+            public TestObj Next { get; set; }
+
+            // Verify we don't eagerly touch all properties
+            public string Fail => throw new NotImplementedException("Don't call this");
+        }
+
+        private static void Add(RecalcEngine engine, string name, object obj)
+        {
+            var objFx = FormulaValue.New(new Wrapper(obj));
+            engine.UpdateVariable(name, objFx);
+        }
+
+        [Theory]        
+        [InlineData("Value(obj.Next.Value)", 20.0)]
+        [InlineData("obj.missing", null)]        
+        [InlineData("IsBlank(obj.Next.Next)", true)]
+        [InlineData("IsBlank(obj.Next)", false)]
+        [InlineData("obj.Next.Next", null)]
+        [InlineData("IsBlank(Index(array, 3))", true)]
```

Comment:
>ta("IsBlank(Index(array, 3))" [](http://example.com/codeflow?start=17&length=29)

You may want to add tests for out of bound index if not already present.

Reply:
added. 
Current behavior is a bit odd: too low is an error, too high is blank.

Reply:
Agree that both should be errors.